File: res/nashorn8.js
var evaluate1 = function () {
    (function () {
        print(eval("this"));
    }).call(this);
};

var evaluate2 = function () {
    var context = {};
    (function () {
        print(eval("this"));
    }).call(context);
};

var evaluate3 = function (context) {
    (function () {
        print(eval("this"));
    }).call(context);
};

File: src/com/winterbe/java8/Nashorn8.java
package com.winterbe.java8;

import jdk.nashorn.api.scripting.NashornScriptEngine;

import javax.script.ScriptEngineManager;
import javax.script.ScriptException;

/**
 * @author Benjamin Winterberg
 */
public class Nashorn8 {
    public static void main(String[] args) throws ScriptException, NoSuchMethodException {
        NashornScriptEngine engine = (NashornScriptEngine) new ScriptEngineManager().getEngineByName("nashorn");
        engine.eval("load('res/nashorn8.js')");

        engine.invokeFunction("evaluate1");
        engine.invokeFunction("evaluate2");
        engine.invokeFunction("evaluate3", "Foobar");
        engine.invokeFunction("evaluate3", new Person("John", "Doe"));
    }

}
